### PR TITLE
[opentitantool] Support HyperDebug transport, configuration files, H1D3 flash chip

### PR DESCRIPTION
**Pull request by jesultra**
Add support for recognizing HyperDebug USB device, with some GPIO and UART support (not complete, need Transport trait to be enhanced to support multiple UART ports.)
Propagate configuration file contents along with Transport to each Command implementation.
Add support for flash chips using other protocols besides the one implemented in struct SpiFlash.

These features are the first steps on the path to Chrome OS using OpenTitan tool in the automated test lab.

### CODE REVIEW

File: sw/host/opentitanlib/src/transport/hyperdebug/uart.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    port: Box<dyn serialport::SerialPort>,
+}
+
+impl HyperdebugUart {
+    pub fn open(hyperdebug: &Hyperdebug) -> Result<Self> {
+        let port =
+            serialport::new(hyperdebug.uart_ttys[0].to_str().ok_or(Error::UnicodePathError)?,
+                            115_200)
+            .timeout(Duration::from_millis(100))
+            .open().expect("Failed to open port");
```
Comment (mcy) [inline sw/host/opentitanlib/src/transport/hyperdebug/uart.rs:25]: Shouldn't this be returned as a Result with ?

File: sw/host/opentitanlib/src/transport/hyperdebug/uart.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        self.port.set_timeout(Duration::MAX)?;
+        self.port.read(buf)
+    }
+}
+
+impl Uart for HyperdebugUart {
+    fn get_baudrate(&self) -> u32 {
+        match self.port.baud_rate() {
+            Ok(baud_rate) => baud_rate,
+            _ => panic!("SerialPort::baud_rate() returned Err")
```
Comment (mcy) [inline sw/host/opentitanlib/src/transport/hyperdebug/uart.rs:54]: maybe `expect()` or `unwrap()` is what you want here? Seems a bit odd to panic, though.

File: sw/host/opentitanlib/src/transport/hyperdebug/uart.rs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+}
+
+impl Write for HyperdebugUart {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        self.port.write(buf)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
```
Comment (mcy) [inline sw/host/opentitanlib/src/transport/hyperdebug/uart.rs:39]: You may want to forward flush() too.

**Comment by cfrantz** [comment]
Lets split this into several PRs:

1. App-config stuff (app/conf.rs).
2. Flash stuff (we should think about the `Flash` trait and whether it would apply to I2C eeproms in the future.
3. Hyperdebug implementation.

**Comment by jesultra** [comment]
This has been split into several other PRs.

**Pull request closed without merging**